### PR TITLE
[pvr] bulk commit queries across the deletion of multiple epgs

### PR DESCRIPTION
**Pull request by phunkyfish**
## Description

Follow on from https://github.com/xbmc/xbmc/pull/18700 which reduces the deletion time by an order of magnitude.

This should reduce the deletion time by the same again.

## How Has This Been Tested?

On OSX using pvr.vuplus and pvr.iptvsimple.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phunkyfish** [comment]
@basilgello can you test to see if this brings the removal time from minutes to seconds?

@ksooo when you have some time to review. This was not as complex as I thought it would be to implement.

File: xbmc/pvr/epg/EpgContainer.cpp [hunk 1]
```diff
@@ -760,7 +766,7 @@ bool CPVREpgContainer::UpdateEPG(bool bOnlyPending /* = false */)
     progressHandler->DestroyProgress();
 
   for (const auto& epg : invalidTables)
-    DeleteEpg(epg);
+    DeleteEpg(epg, true);
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgContainer.cpp:769]: @phunkyfish  do we have a chance to get rid of this (last) sync delete?

Comment (phunkyfish) [reply xbmc/pvr/epg/EpgContainer.cpp:769]: Sorry, yes, that's an obvious change. I didn't see it at all!

Comment (phunkyfish) [reply xbmc/pvr/epg/EpgContainer.cpp]: Ok, latest push should add that remaining case.

File: xbmc/pvr/epg/EpgContainer.cpp [hunk 2]
```diff
@@ -759,8 +755,17 @@ bool CPVREpgContainer::UpdateEPG(bool bOnlyPending /* = false */)
...
     progressHandler->DestroyProgress();
 
+  database->Lock();
   for (const auto& epg : invalidTables)
-    DeleteEpg(epg);
+  {
+    QueueDeleteEpg(epg);
+
+    size_t queryCount = database->GetDeleteQueriesCount();
+    if (queryCount > 10000)
```
Comment (ksooo) [inline xbmc/pvr/epg/EpgContainer.cpp:764]: I feel we should define a constant in EpgDateBase.h and PVRDatabase.h for this magic number now that is is used all over the place.

File: xbmc/pvr/channels/PVRChannelGroupInternal.cpp [hunk 3]
```diff
@@ -285,24 +286,38 @@ std::vector<std::shared_ptr<CPVRChannel>> CPVRChannelGroupInternal::RemoveDelete
   bool channelsDeleted = false;
 
   const std::shared_ptr<CPVRDatabase> database = CServiceBroker::GetPVRManager().GetTVDatabase();
-  if (!database)
+  const auto epgDatabase = CServiceBroker::GetPVRManager().EpgContainer().GetEpgDatabase();
```
Comment (ksooo) [inline xbmc/pvr/channels/PVRChannelGroupInternal.cpp:289]: Could you do me a favor and not use auto here. I prefer to use auto only for more complex types than shared pointers.

File: xbmc/pvr/epg/EpgContainer.cpp (hunk 2 above)
Comment (phunkyfish) [reply xbmc/pvr/epg/EpgContainer.cpp:764]: Done

File: xbmc/pvr/channels/PVRChannelGroupInternal.cpp (hunk 3 above)
Comment (phunkyfish) [reply xbmc/pvr/channels/PVRChannelGroupInternal.cpp:289]: Done

**Comment by phunkyfish** [comment]
@ksooo, comments addressed, should be ready now.

**Review by ksooo — APPROVED** [review]
lgtm, thx

**Pull request merged**